File: LeetCode/66 - Plus One/solution.py
class Solution:
    def plusOne(self, digits: list[int]) -> list[int]:
        # convert the list of integers to a list of strings
        new_digits: list[str] = [str(i) for i in digits]

        # convert the list of string numbers to a single integer
        number: int = int("".join(new_digits))

        # increment by 1
        number += 1

        # convert the integer to a string then convert each back into an integer as part of a list
        answer: list[int] = [int(d) for d in str(number)]

        return answer

File: LeetCode/66 - Plus One/solution.swift
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        // string list version of the digits array
        var new_digits: [String] = []

        // integer version of the new number
        var number: Int

        // integer list version of the new number
        var answer: [Int] = []

        for i: Int in digits {
            new_digits.append(String(i))
        }

        // convert the string list from above to an integer value incremented by 1
        number = Int(new_digits.joined(separator: ""))! + 1



    }
}